src/main/java/commandparser/CommandParser.java

```diff
@@ -0,0 +1,166 @@
+                response.append(counter).append(". ").append(task.toString()).append("\n");
+                counter++;
+            }
+        }
+
+        if (counter == 1) {
+            return "No matching tasks found.";
+        }
+
+        return response.toString();
+    }
+
+    /**
+     * Adds a deadline task based on user input.
+     *
+     * @param input The user input containing the task description and deadline.
+     * @param isSilent If true, suppress output messages.
+     */
+    public String handleDeadline(String input, boolean isSilent) {
+        if (!input.contains("/by ")) {
+            return "You need a deadline to add this task!";
+
+        }
+        String[] parts = input.split("/by ");
+        String taskName = parts[0].replaceFirst("deadline ", "").trim();
+        if (taskName.isEmpty()) {
+            return "You need a task description!";
+
+        }
+        String initial = parts[1].trim();
+        LocalDate deadline = LocalDate.parse(initial);
+        Deadline deadlineTask = new Deadline(taskName, deadline, input);
+        String temp = taskManager.addTask(deadlineTask, isSilent);
+        this.storage.writeTasks();
+        return temp;
+    }
+
+    /**
+     * Adds a todo task based on user input.
+     *
+     * @param input The user input containing the task description.
+     * @param isSilent If true, suppress output messages.
+     */
+    public String handleTodo(String input, boolean isSilent) {
+        String taskName = input.replaceFirst("todo ", "").trim();
+        if (taskName.isEmpty()) {
+            return "You need a task description!";
+        }
+        Todo todoTask = new Todo(taskName, input);
+        String temp = taskManager.addTask(todoTask, isSilent);
+        this.storage.writeTasks();
+        return temp;
+    }
+
+    /**
+     * Adds an event task based on user input.
+     *
+     * @param input The user input containing the task description, start, and end time.
+     * @param isSilent If true, suppress output messages.
+     */
+    public String handleEvent(String input, boolean isSilent) {
+        if (!input.contains("/from ") || !input.contains("/to ")) {
+            return "You need a starting and ending date to add this task!";
+
+        }
+        String[] parts = input.split("/from ");
+        String[] dateParts = parts[1].split("/to ");
+        String taskName = parts[0].replaceFirst("event ", "").trim();
+        if (taskName.isEmpty()) {
+            return "You need a task description!";
+
+        }
+        String initialStartDate = dateParts[0].trim();
+        String initialEndDate = dateParts[1].trim();
+        LocalDate startDate = LocalDate.parse(initialStartDate);
+        LocalDate endDate = LocalDate.parse(initialEndDate);
+        Event eventTask = new Event(taskName, startDate, endDate, input);
+        String temp = taskManager.addTask(eventTask, isSilent);
+        this.storage.writeTasks();
+        return temp;
+    }
+}
```

Comment:
insert blankline after if block and use blank line to separate different logic blocks